libsolidity/interface/FileAndSMTCallbacks.h

```diff
@@ -0,0 +1,54 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+#pragma once
+
+#include <libsolidity/interface/FileReader.h>
+#include <libsolidity/interface/SMTSolverCLI.h>
+
+namespace solidity::frontend
+{
+
+class FileAndSMTCallbacks
```

Comment:
It would be best to consistently refer to the `FileReader`'s callback as "import callback".
```suggestion
class ImportAndSMTCallbackAdapter
```
or how about we just call it an "universal callback"?
```suggestion
class UniversalCallback
```

Reply:
Changed to `UniversalCallback`.